feat: show progress bar while fetching marks

Sami-GT opened the pull request:
#### New feature
- Add a progress bar that updates as marks are fetched.
- Display the number of students processed out of the total.
#### Reason for adding this feature
Fetching marks from the website could take a long time and users don't see any indication of progress. This may lead to the impression that the process is stuck.
#### Closes
Closes #5

Sami-GT committed:
feat: show progress bar while fetching marks

Sami-GT commented:
> #### New feature
> * Add a progress bar that updates as marks are fetched.
> * Display the number of students processed out of the total.
> 
> #### Reason for adding this feature
> Fetching marks from the website could take a long time and users don't see any indication of progress. This may lead to the impression that the process is stuck.
> 
> #### Closes
> Closes #5

ObaidaNa reviewed (commented):
Thanks for your contribution
here's some points to do:
- You have missed the point of using asynchronous requests processing for handling i/o bounds tasks, You should turn back this:
```python
tasks = [
            asyncio.create_task(one_req(int(number), session, recurse_limit))
            for number in numbers
        ]
```
by this, all requests will be sent at the same time, for processing the progress bar, I would suggest to make a callback function that will called by the `one_req` function, so when the task is done, it will call the callback function to update the progress message.

- secondly, I would rather to split the logic of calculating and updating the message in a separated function as it will be cleaner and nicer, maybe more reusable, and applying the single responsibility principle

Sami-GT committed:
refactor: asynchrounous requests and improve progress handling

- Added a callbcak function for updating the progress message upon task completion.
- Split logic for calculating and updating the progress message into separate functions

Sami-GT commented:
> * You have missed the point of using asynchronous requests

You're right. I tested its speed, and it was fast enough, but doing it the way you suggested is definitely better, so I followed your advice. Thanks!

ObaidaNa reviewed (approved):
Nice work Sami, Thanks
